balancer-js/src/modules/exits/exits.module.ts

```diff
@@ -699,6 +751,113 @@ export class Exit {
     };
   }
 
+  private createExitPoolProportional(
+    node: Node,
+    minAmountsOut: string[],
+    sender: string,
+    recipient: string
+  ): {
+    modelRequest: ExitPoolModelRequest;
+    encodedCall: string;
+    bptIn: string;
+    tokensOut: string[];
+    amountsOut: string[];
+  } {
+    const isRootNode = !node.parent;
+    const amountIn = isRootNode
+      ? node.index
+      : Relayer.toChainedReference(this.getOutputRef(0, node.index)).toString();
+
+    const tokensOut = node.children.map((child) => child.address);
+    const amountsOut = [...minAmountsOut];
+
+    if (node.type === PoolType.ComposableStable) {
+      // assets need to include the phantomPoolToken
+      tokensOut.push(node.address);
+      // need to add a placeholder so sorting works
+      amountsOut.push('0');
+    }
+
+    // TODO: we shoule consider let the graph handle sorting instead of manipulating
+    // token order within actions - specially now that we have different sorting
+    // cases and that the subgraph is already handling them properly
+
+    // sort inputs
+    const assetHelpers = new AssetHelpers(this.wrappedNativeAsset);
+    const [sortedTokens, sortedAmounts] = assetHelpers.sortTokens(
+      tokensOut,
+      amountsOut
+    ) as [string[], string[]];
+
+    let userData: string;
+    if (node.type === PoolType.Weighted) {
+      userData = WeightedPoolEncoder.exitExactBPTInForTokensOut(amountIn);
+    } else {
+      userData = StablePoolEncoder.exitExactBPTInForTokensOut(amountIn);
```

Comment:
Is this correct? I thought the encoding is different for ComposableStables. (Latest `VaultActions.sol` in mono-repo is quite a good reference)